src/app/_components/ManageAttestation.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+import { NextPage } from "next";
+import useGetAttestations from "@/hooks/useGetAttestations";
+import {
+  Table,
+  TableCaption,
+  TableHeader,
+  TableRow,
+  TableHead,
+  TableBody,
+  TableCell,
+} from "@/components/ui/table";
+import RevokeAttestationPage from "./RevokeAttestation";
+
+const ManageAttestation: NextPage = () => {
+  const { attestations, loading, error } = useGetAttestations();
+  console.log(attestations);
+
+  const renderTableHeader = () => {
+    if (attestations && attestations.length > 0) {
+      return (
+        <TableRow>
+          <TableHead className="font-semibold">Action</TableHead>
+          {Object.keys(attestations[0]).map((key) => (
+            <TableHead className="font-semibold" key={key}>
+              {formatKey(key)}
+            </TableHead>
+          ))}
+        </TableRow>
+      );
+    }
+    return null;
+  };
+
+  const renderTableRows = () => {
+    return attestations?.map((attestation: any) => (
+      <TableRow key={attestation.id}>
+        <TableCell className="text-right">
+          <RevokeAttestationPage
+            uid={attestation.id}
+            schemaId={attestation.schema.id}
+          />
+        </TableCell>
+        {Object.entries(attestation).map(([key, value], index) => (
+          <TableCell
+            key={index}
+            className={
+              index === Object.entries(attestation).length - 1
+                ? "text-right"
+                : ""
+            }
+          >
+            {key === "revocationTime" || key === "expirationTime"
+              ? formatTime(Number(value), key)
+              : key === "schema"
+                ? (value as any).id // Ensure value is treated as an object
+                : String(value)}
+          </TableCell>
+        ))}
+      </TableRow>
+    ));
+  };
+
+  const formatKey = (key: string): string => {
+    // Replace underscores with spaces and split camelCase words with a space
+    const result = key
+      .replace(/([A-Z])/g, " $1") // Insert space before each uppercase letter
+      .replace(/_/g, " ") // Replace underscores with spaces
+      .trim(); // Remove leading and trailing spaces
+
+    // Capitalize the first letter of each word
+    return result
+      .split(" ")
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+      .join(" ");
```

Comment:
`#could` be defined outside the component, e.g. as part of `lib/` if we use this elsewhere?

Reply:
sure will move to it lib

Reply:
done